main.go

```diff
@@ -557,6 +557,13 @@ func checkOneInitialCollectionStatus(ctx context.Context, server *state.Server,
 	if err != nil {
 		return err
 	}
+
+	// We don't need a mutex here, because we only check once at startup
+	dataDirectorySetting := postgres.GetDataDirectory(server, settings)
+	if dataDirectorySetting != "" && server.Config.DataDirectory == "" {
+		server.Config.DataDirectory = dataDirectorySetting
+	}
```

Comment:
Technically this could also be written as:

```golang
	if server.Config.DataDirectory == "" {
		server.Config.DataDirectory = postgres.GetDataDirectory(server, settings)
	}
```